src/index.ts

```diff
@@ -6,30 +6,77 @@ import { IThemeManager } from '@jupyterlab/apputils';
 import { themeEditorIcon } from './icons';
 import { ThemeEditorModel } from './model';
 import { ThemeEditorView } from './view';
-import { IChangedArgs } from '@jupyterlab/coreutils';
+import { ISettingRegistry } from '@jupyterlab/settingregistry';
+
+const PLUGIN_ID = 'jupyter-theme-editor:plugin';
 
 /**
  * Initialization data for the jupyter-theme-editor extension.
  */
 const plugin: JupyterFrontEndPlugin<void> = {
-  id: 'jupyter-theme-editor:plugin',
+  id: PLUGIN_ID,
   autoStart: true,
   requires: [IThemeManager],
-  activate: (app: JupyterFrontEnd, themeManager: IThemeManager) => {
-    const onThemeChanged = (
-      themeManager: IThemeManager,
-      changes: IChangedArgs<string, string | null, string>
-    ) => {
+  optional: [ISettingRegistry],
+  activate: (
+    app: JupyterFrontEnd,
+    themeManager: IThemeManager,
+    settingRegistry: ISettingRegistry
+  ) => {
+    const onThemeChanged = (themeManager: IThemeManager) => {
       themeManager.themeChanged.disconnect(onThemeChanged);
-
       const model = new ThemeEditorModel();
       const view = new ThemeEditorView(model);
       view.addClass('jp-theme-editor-view-panel');
       view.id = 'theme-editor';
       view.title.icon = themeEditorIcon;
       app.shell.add(view, 'left');
-
       console.log('JupyterLab extension jupyter-theme-editor is activated!');
+
+      /**
+       * Load the settings for this extension
+       *
+       * @param setting Extension settings
+       */
+
+      function loadSetting(
+        setting: ISettingRegistry.ISettings
+      ): [boolean, any] {
+        // Read the settings and convert to the correct type
+        const useSettings = setting.get('useSettings').composite as boolean;
+        const customStyles = setting.get('customStyles').composite as any;
+        return [useSettings, customStyles];
+      }
+
+      Promise.all([app.restored, settingRegistry.load(PLUGIN_ID)]).then(
+        ([, setting]) => {
+          // Read the settings
+          setting.changed.connect(loadSetting);
+          const [useSettings, customStyles] = loadSetting(setting);
+          const onModelChanged = () => {
+            console.log('Model has changed');
+            let cssValue: string;
+
+            for (const key of Object.keys(model.formData)) {
+              if (typeof model.formData[key] === 'string') {
+                cssValue = model.formData[key].trim();
+                model.formData[key] = cssValue;
+              }
+              customStyles[key] = model.formData[key];
+              setting.set('customStyles', customStyles);
+            }
+          };
+
+          if (useSettings === false) {
```

Comment:
This is not working due to the scope of your variable.

`useSettings` is defined in line 55 and it is independent of `useSettings` defined line 46; they are both local to their functions.

If you want `useSettings` to be changed every time `setting.changed` is emitted, you need to define your variable `useSettings` in a parent scope of both `loadSetting` and the function line 52.

A logic like the following:

```typescript
let a = 0;

const foo = () => {
  a = a + 1;
}

const bar = () => {
  console.log(a);
}
```